### ISSUE TITLE
Parsing a `return macro @:pos` crashes the parser

### ISSUE THREAD

**Issue by dubspeed** [HaxeCheckstyle/haxe-rename#8]
👋 Hello, 

given a macro like this:

```
class TestCrashMacro {
	static function combine(structure:Expr):Expr {
		if (structure.expr.match(EDisplay(_, DKMarked))) {
			return macro @:pos(Context.currentPos()) ($structure : {});
		}
        return Expr();
	}
}
```

will crash rename.js when parsing `@:pos`.

haxe_ValueException [Error]: failed to parse src/SecondRenameCrash.hx - Exception: invalid token tree structure - found:[@]
Called from tokentree_walk_WalkTypedefBody.walkTypedefCurlyBody (rename.js line 12750 column 26)

**Comment by AlexHaxe** [comment]
if you have more of those crashes, I would like to prioritise them, since most likely they are things that break tokentree which affects a lot more than just renaming.

**Comment by AlexHaxe** [comment]
even though I haven't made a release or commit to haxe-rename lib, both this and the crash from https://github.com/HaxeCheckstyle/haxe-rename/issues/6 should be fixed in vshaxe 2.32.0 and above, since they come with an updated tokentree library
